rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -69,21 +69,43 @@
 
 namespace
 {
-
 //==============================================================================
-// A function to take ros topic names and convert them to valid Zenoh keys.
+// A function that generates a key expression for message transport of the format
+// <ros_domain_id>/<topic_name>/<topic_type>/<topic_hash>
 // In particular, Zenoh keys cannot start or end with a /, so this function
 // will strip them out.
 // The Zenoh key is also prefixed with the ros_domain_id.
 // Performance note: at present, this function allocates a new string and copies
 // the old string into it. If this becomes a performance problem, we could consider
 // modifying the topic_name in place. But this means we need to be much more
 // careful about who owns the string.
-z_owned_keyexpr_t ros_topic_name_to_zenoh_key(const char * const topic_name, size_t domain_id)
-{
-  const std::string keyexpr_str = std::to_string(domain_id) +
-    "/" +
-    rmw_zenoh_cpp::liveliness::mangle_name(topic_name);
+z_owned_keyexpr_t ros_topic_name_to_zenoh_key(
+  const std::size_t domain_id,
+  const char * const topic_name,
+  const char * const topic_type,
+  const char * const type_hash)
+{
+  auto strip_slashes =
+    [](const char * const str) -> std::string
+    {
+      std::string ret = std::string(str);
+      const std::size_t len = strlen(str);
+      std::size_t start = 0;
+      std::size_t end = len - 1;
+      if (str[0] == '/') {
+        ++start;
+      }
+      if (str[end] == '/') {
+        --end;
+      }
+      return ret.substr(start, end - start);
+    };
```

Comment:
I think it would be nicer to make this a separate function, rather than a lambda here.

Reply:
Done https://github.com/ros2/rmw_zenoh/pull/171/commits/5796e7426454f4ac73ce37a28a00954cf327c32c